plugins/directory-listing/src/main/java/com/blazemeter/jmeter/DirectoryIterator.java

```diff
@@ -0,0 +1,73 @@
+package com.blazemeter.jmeter;
+
+
+import org.apache.jmeter.threads.JMeterContextService;
+
+import java.io.File;
+import java.util.Iterator;
+import java.util.List;
+
+public class DirectoryIterator {
+
+    private final DirectoryListingConfig config;
```

Comment:
bad decision. business logic class should not depend on its clients. It's client class that should depend on business logic class.
The solution is to make DirectoryIterator to hold all the flags as fields. Or have dedicated Configuration class. Or have bitmasks to not have too many methods and fields.